src/Container.ts

```diff
@@ -89,8 +92,16 @@ export class Container {
             params.unshift(null);
         }
 
-        // create a new instance of the requested object
-        const objectInstance = new (type.bind.apply(type, params))();
+        let objectInstance: any;
+
+        if (factory) {
+            // Calling factory function if specified to create an instance.
+            objectInstance = factory.apply(undefined, params);
```

Comment:
Could you please provide an example or a an edge case, usage scenario? My imagination is failing me here.